src/Tool/MacAuthorizationTrait.php

```diff
@@ -26,7 +26,7 @@
     /**
      * Returns the id of this token for MAC generation.
      *
-     * @param  AccessToken $token
+     * @param  AccessToken|null $token
```

Comment:
Dropping the `= null` default makes sense to me. [This condition already prevents it](https://github.com/thephpleague/oauth2-client/blob/1b0fe447ae474bcfd83f3f8578ce45f5077f2630/src/Provider/AbstractProvider.php#L811-L816) so we lose nothing.